R/utils.R

```diff
@@ -0,0 +1,91 @@
+#' decode_url_key
+#'
+#' @description Decode the key in the url to get a password for the access
+#'   token.
+#'
+#' @param hash_key Key that is given to decrypt
+#' @param url_key Encrypted key retrieved from the url
+#' @param gunzip Whether to gunzip the key after decoding.
+#' @param debug Whether to enable debugging messages.
+#'
+#' @return String containing the encrypted password which can be used to get an
+#'   access token
+#'
+#' @export
+#' @importFrom glue glue
+#' @importFrom jsonlite fromJSON
+#' @importFrom logr log_print
+#' @importFrom openssl aes_cbc_decrypt base64_decode
+#' @importFrom utils URLdecode
+decode_url_key <- function(hash_key, url_key, gunzip = FALSE, debug = FALSE) {
+
+  # Replace underscores and add padding if key is gzipped
+  if (gunzip) {
+    url_key <- chartr("-_", "+/", url_key)
+    url_key <- paste0(url_key, strrep("=", (4 - nchar(url_key) %% 4) %% 4))
+  }
+
+  # Decode key
+  key_raw <- openssl::base64_decode(URLdecode(url_key))
+
+  # Unzip string if compressed
+  if (gunzip) {
+    key_raw <- memDecompress(key_raw, type = "gzip")
+  }
+
+  # Decode key
+  key_decoded <- rawToChar(key_raw)
+
+  if (debug) {
+    logr::log_print(
```

Comment:
Is het nodig om hier de packagenaam voor de functie te zetten?

Reply:
Waarschijnlijk niet. Kan zijn dat deze functie toevallig in meerdere packages voorkomt, maar dat denk ik hier eigenlijk niet.